src/components/ObsDetails/DetailsTab/DQAVoteButtons.js

```diff
@@ -8,46 +8,60 @@ import {
 import { View } from "components/styledComponents";
 import * as React from "react";
 import { useTheme } from "react-native-paper";
-import { useTranslation } from "sharedHooks";
+import { useCurrentUser, useTranslation } from "sharedHooks";
 
 type Props = {
   metric: string,
-  qualityMetrics: Object,
+  votes: [Object],
   loadingAgree: boolean,
   loadingDisagree: boolean,
   loadingMetric: ?string,
   setVote: Function,
   removeVote: Function
 }
 
-const getUserVote = ( metric, qualityMetrics ) => {
-  if ( qualityMetrics ) {
-    const match = qualityMetrics.find( element => (
-      element.metric === metric && element.user_id ) );
+const getUserVote = ( currentUser, metric, votes ) => {
+  if ( votes && votes.length > 0 ) {
+    const match = votes.find( element => ( element.user_id === currentUser?.id ) );
```

Comment:
@angielt when I checked the schema of the `Vote`s that are returned for the dq metric "needs_id" from the server and used here from what I can tell all have a user_id which is just the id of the user who created the Vote. So, I changed the check here to find the element that is from the current user. so that the function checks wether the current user has voted up or down. The schema for the `QualityMetric` has the same structure with user_id, so I think it applies also to the other quality metrics. Let me know if you see anything off with this change.

Reply:
Good catch, I don't see anything off with this change. I didn't know about the groupBy function in lodash until now. It seems very handy for the DQA.